third_party/gif_encoder/src/main/java/com/bumptech/glide/gifencoder/AnimatedGifEncoder.java

```diff
@@ -252,15 +278,30 @@ public void setQuality(int quality) {
      *          int frame width.
      */
     public void setSize(int w, int h) {
-        if (started && !firstFrame)
+        if (started)
             return;
+
+        fixedWidth = w;
+        fixedHeight = h;
+        if (fixedWidth < 1)
+            fixedWidth = 320;
+        if (fixedHeight < 1)
+            fixedHeight = 240;
+
+        sizeSet = true;
+    }
+
+    /**
+     * Sets current GIF frame size.
```

Comment:
Would you mind adding a clarifying comment about how this differs from setSize and when to use one vs the other?